### PR TITLE
cephfs: adjust RBAC for CephFS provisioner role

### PR DESCRIPTION
**Pull request by humblec**
CephFS CSI driver does not need permissions on Node,ConfigMap objects. 

Updates: #3018

### CODE REVIEW

File: deploy/cephfs/kubernetes/csi-provisioner-rbac.yaml [hunk 1]
```diff
@@ -30,25 +27,25 @@ rules:
     verbs: ["get", "list", "watch"]
   - apiGroups: ["snapshot.storage.k8s.io"]
     resources: ["volumesnapshots"]
-    verbs: ["get", "list"]
+    verbs: ["get", "list", "watch"]
```
Comment (nixpanic) [inline deploy/cephfs/kubernetes/csi-provisioner-rbac.yaml]: why is this added? explain it in the commit message please

File: deploy/cephfs/kubernetes/csi-provisioner-rbac.yaml [hunk 2]
```diff
@@ -30,25 +27,25 @@ rules:
...
     resources: ["volumeattachments"]
-    verbs: ["get", "list", "watch", "update", "patch"]
+    verbs: ["get", "list", "watch", "patch"]
   - apiGroups: ["storage.k8s.io"]
     resources: ["volumeattachments/status"]
     verbs: ["patch"]
   - apiGroups: [""]
     resources: ["persistentvolumeclaims/status"]
-    verbs: ["update", "patch"]
+    verbs: ["patch"]
```
Comment (nixpanic) [inline deploy/cephfs/kubernetes/csi-provisioner-rbac.yaml]: does this require a certain version of other Kubernetes components?

File: deploy/cephfs/kubernetes/csi-provisioner-rbac.yaml [hunk 3]
```diff
@@ -78,9 +75,6 @@ metadata:
   name: cephfs-external-provisioner-cfg
 rules:
   # remove this once we stop supporting v1.0.0
```
Comment (nixpanic) [inline deploy/cephfs/kubernetes/csi-provisioner-rbac.yaml]: does this comment not refer to the ConfigMaps section? if that is the case, remove it as well

**Review by nixpanic — DISMISSED** [review]
instead of the comment that these are 'unwanted', a more detailed explanation would be much appreciated in the commit message. Were these never needed, or have components changed to use a new API (like update -> patch), which should mention the version of the component where the change was done.

File: deploy/cephfs/kubernetes/csi-provisioner-rbac.yaml (hunk 2 above)
Comment (humblec) [reply deploy/cephfs/kubernetes/csi-provisioner-rbac.yaml]: all supported versions of kube should work here.

File: deploy/cephfs/kubernetes/csi-provisioner-rbac.yaml (hunk 3 above)
Comment (humblec) [reply deploy/cephfs/kubernetes/csi-provisioner-rbac.yaml]: yep, removed that as well.

File: deploy/cephfs/kubernetes/csi-provisioner-rbac.yaml (hunk 1 above)
Comment (humblec) [reply deploy/cephfs/kubernetes/csi-provisioner-rbac.yaml]: not needed.. removed.

**Comment by humblec** [comment]
@Madhu-1 ptal.. thanks..

**Review by Madhu-1 — APPROVED** [review]

**Review by yati1998 — DISMISSED** [review]

**Comment by humblec** [comment]
@Mergifyio refresh

**Comment by humblec** [comment]
@Mergifyio rebase

File: deploy/cephfs/kubernetes/csi-provisioner-rbac.yaml [hunk 4]
```diff
@@ -36,19 +33,19 @@ rules:
...
     resources: ["volumesnapshotcontents"]
-    verbs: ["create", "get", "list", "watch", "update", "delete", "patch"]
+    verbs: ["get", "list", "watch", "update", "patch"]
   - apiGroups: ["snapshot.storage.k8s.io"]
     resources: ["volumesnapshotclasses"]
     verbs: ["get", "list", "watch"]
   - apiGroups: ["storage.k8s.io"]
     resources: ["volumeattachments"]
-    verbs: ["get", "list", "watch", "update", "patch"]
+    verbs: ["get", "list", "watch", "patch"]
```
Comment (Madhu-1) [inline deploy/cephfs/kubernetes/csi-provisioner-rbac.yaml]: as the discussion is going on in the kubernetes-csi repo. keeping this on hold as well. https://github.com/kubernetes-csi/external-attacher/pull/354

**Review by Madhu-1 — CHANGES_REQUESTED** [review]

File: deploy/cephfs/kubernetes/csi-provisioner-rbac.yaml (hunk 4 above)
Comment (humblec) [reply deploy/cephfs/kubernetes/csi-provisioner-rbac.yaml]: @Madhu-1 I have removed the update/patch for now. please revisit

**Comment by humblec** [comment]
@Madhu-1 can you bring back the the approval?

**Review by Madhu-1 — APPROVED** [review]

**Review by yati1998 — APPROVED** [review]
